### ISSUE TITLE
linter: CLI --allow category disables rules explicitly enabled in config

### ISSUE THREAD

**Issue by yoshi-taka** [oxc-project/oxc#21965]
### What version of Oxlint are you using?

1.61.0

### What happened?

# linter: CLI --allow category disables rules explicitly enabled in config

## Summary

When using `oxlint` with a config file, rules explicitly enabled in the config (e.g. `no-restricted-imports`) appear to be disabled when `--allow` is used to restrict categories from the CLI.

This makes it difficult to run only a small subset of rules in large repositories.

---

## Environment

- oxlint version: v1.62

---

## What I want to achieve

- Keep specific rules enabled via config (e.g. `no-restricted-imports`)
- Disable all other default/category rules via CLI
- Effectively run only a small subset of rules for performance reasons

---

### Config (`config.json`)

```json
{
  "rules": {
    "no-restricted-imports": [
      "warn",
      {
        "paths": [
          { "name": "@material-ui/core" }
        ]
      }
    ]
  }
}
```

### Case 1: Works as expected

```sh
oxlint --import-plugin -D no-restricted-imports -c config.json src/App.js
```

→ `no-restricted-imports` is reported

---

### Case 2: Does NOT work

```sh
oxlint --import-plugin -A all -D no-restricted-imports -c config.json src/App.js
```

→ No report

---

### Case 3: Does NOT work

```sh
oxlint --import-plugin \
  -A correctness -A suspicious -A pedantic -A perf -A style -A restriction \
  -D no-restricted-imports \
  -c config.json src/App.js
```

→ No report

---

## Observed behavior

Using `--allow` categories from CLI appears to suppress rules even if they are explicitly enabled in the config file.

---

## Expected behavior

One of the following:

1. Config-enabled rules should remain active regardless of CLI `--allow`
2. Or, there should be a documented precedence rule explaining this behavior clearly

---

## Why this matters

In large repositories, we often want to run only a small number of rules (e.g. import restrictions) for targeted analysis.

Currently, disabling categories via CLI also disables config-enabled rules, which makes it difficult to:

- control performance cost  
- embed oxlint as a selective analysis tool  
- run minimal rule sets in a single process  

---

## Related notes

I noticed recent fixes around category / plugin handling:

- #21521  
- #19411  

However, this issue seems to affect top-level config + CLI interaction, not overrides.

---

## Request

If this is intentional behavior, it would be very helpful to have:

- a "config-only" mode (run only rules defined in config), or  
- a way to explicitly prevent CLI category filtering from disabling config rules  

Thank you!

**Comment by aartisonigra** [comment]
Hi @yoshi-taka , I'd like to work on this issue. Could you please assign it to me?"

**Comment by aartisonigra** [comment]
Hi @yoshi-taka , I have submitted this PR to fix the issue where CLI category filters were overriding explicit config rules. All checks are passing. Could you please take a look when you have a moment? Thank you!



Description
This PR fixes a bug where CLI category filters (like --allow correctness) were overriding rules that the user had explicitly enabled in their configuration file.

Changes
Updated upsert_where function in config_builder.rs.
Replaced .insert() with .entry().or_insert() to preserve existing rule settings.
Verification
Successfully ran cargo check.
Verified that rules from the config file are no longer disabled when using category filters in the CLI.

**Comment by yoshi-taka** [comment]
Hi @aartisonigra , thanks for your interest in this issue!

I’m not a maintainer of the project, so I’m not able to assign issues.  
I saw that you’ve already opened a PR. Feel free to link this issue in the PR description if relevant.

Thanks!
